packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/index.tsx

```diff
@@ -0,0 +1,238 @@
+
+export type QuantityApplicationRateProps = {
+  productId: number | string;
+  isReadOnly: boolean;
+  system: 'metric' | 'imperial';
+  location: Location;
+  defaultValues?: TaskProductFormFields;
+};
+
+const QuantityApplicationRate = ({
+  productId,
+  isReadOnly,
+  system, // measurementSelector
+  location,
+  defaultValues,
+}: QuantityApplicationRateProps) => {
+  const { t } = useTranslation();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpanded = () => setIsExpanded((prev) => !prev);
+
+  const { total_area, total_area_unit, type } = location;
+
+  const { control, getValues, setValue, register, watch } = useForm<TaskProductFormFields>({
+    defaultValues: {
+      [TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION_AMENDED]: 100,
+      ...defaultValues,
+    },
+  });
+
+  /* Control of relationship between quantity, area, and application rate */
+  const {
+    onQuantityChange,
+    onApplicationRateChange,
+    onPercentLocationChange,
+    isWeight,
+    toggleMeasure,
+    previewStringValue,
+    previewStringUnit,
+  } = useQuantityApplicationRate({
+    total_area,
+    total_area_unit,
+    system,
+    watch,
+    setValue,
+    getValues,
+  });
+
+  const percent_of_location = watch(TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION_AMENDED);
+
+  /* Label for weight/volume switch */
+  const formatLabel = (labelKey: string, shouldBeBold: boolean) => {
+    // t('ADD_TASK.SOIL_AMENDMENT_VIEW.WEIGHT')
+    // t('ADD_TASK.SOIL_AMENDMENT_VIEW.VOLUME')
+    const label = t(labelKey);
+    return shouldBeBold ? (
+      <span className={styles.switchLabel}>
+        <b>{label}</b>
+      </span>
+    ) : (
+      <span className={styles.switchLabel}>{label}</span>
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <Switch
+        leftLabel={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.WEIGHT', isWeight)}
+        label={formatLabel('ADD_TASK.SOIL_AMENDMENT_VIEW.VOLUME', !isWeight)}
+        isToggleVariant
+        onChange={toggleMeasure}
+        checked={!isWeight}
+      />
+      <div className={styles.applicationRateCard}>
+        <div>
+          {/* @ts-ignore */}
+          <Unit
+            label={t('ADD_TASK.SOIL_AMENDMENT_VIEW.QUANTITY')}
+            data-cy="soilAmendment-quantity"
+            register={register}
+            name={TASK_PRODUCT_FIELD_NAMES[isWeight ? 'WEIGHT' : 'VOLUME']}
+            displayUnitName={TASK_PRODUCT_FIELD_NAMES[isWeight ? 'WEIGHT_UNIT' : 'VOLUME_UNIT']}
+            unitType={SOIL_AMENDMENT_UNITS[isWeight ? 'WEIGHT' : 'VOLUME'].units}
+            system={system}
+            hookFormSetValue={setValue}
+            hookFormGetValue={getValues}
+            hookFromWatch={watch}
+            control={control}
+            mode={'onChange'}
+            onChange={onQuantityChange}
+            disabled={isReadOnly}
+            required
+            key={isWeight ? 'weight' : 'volume'}
+          />
+          <AreaApplicationSummary
```

Comment:
I don't want to be a bummer haha, just wondering if it is going to be reused somewhere or if maybe it is small enough to just put in here.

Reply:
Do you mean as a sub-component in the file or just insert the JSX right in there? I'm okay with either; I write it that way initially and usually extract it only for readability. If it's not more readable I'll stick it back in!

Reply:
no action needed -- but it could probably be jammed right in there if you agree/ want to